packages/oauth2/src/id-token/verify-id-token.ts

```diff
@@ -0,0 +1,93 @@
+import type { CallbackContext } from '../callbacks'
+import { extractJwkFromJwksForJwt } from '../common/jwk/jwks'
+import { decodeJwt } from '../common/jwt/decode-jwt'
+import { verifyJwt } from '../common/jwt/verify-jwt'
+import { Oauth2Error } from '../error/Oauth2Error'
+import type { AuthorizationServerMetadata } from '../metadata/authorization-server/z-authorization-server-metadata'
+import { fetchJwks } from '../metadata/fetch-jwks-uri'
+import { zIdTokenJwtHeader, zIdTokenJwtPayload } from './z-id-token-jwt'
+
+export interface VerifyJwtIdTokenOptions {
+  /**
+   * The compact id token.
+   */
+  idToken: string
+
+  /**
+   * Callbacks used for verifying the id token
+   */
+  callbacks: Pick<CallbackContext, 'verifyJwt' | 'fetch'>
+
+  /**
+   * If not provided current time will be used
+   */
+  now?: Date
+
+  /**
+   * Authorization server metadata
+   */
+  authorizationServer: AuthorizationServerMetadata
+
+  /**
+   * The client_id of the Relying Party for which the token was issued.
+   */
+  clientId: string
+
+  /**
+   * Expected nonce in the payload. If not provided the nonce won't be validated.
+   */
+  expectedNonce?: string
+}
+
+/**
+ * Verify an ID Token JWT.
+ */
+export async function verifyJwtIdToken(options: VerifyJwtIdTokenOptions) {
```

Comment:
I'd call this `verifyIdTokenJwt` as it reads more natural

Reply:
ah it's odd i did it that way.

probably because it's called "Jwt Profile". So it'd be `jwtProfileAccessTokenJwt` which is just weird.